trieste/models/gpflux/__init__.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+This package contains the primary interface for deep Gaussian process models. It also contains a
+number of :class:`TrainableProbabilisticModel` wrappers for GPflux-based models. Note that currently
+copying/saving models is not supported, so in a Bayes Opt loop `track_state` should be set False.
```

Comment:
What happens if it's True? Will the user at least see the exception that clearly explains what's going on?

Reply:
I think it crashes in a rather messy way, but Hrvoje and I have decided we'll leave this for an issue.

Reply:
Perhaps we could add a runtime check in `optimize`? Or better yet overload `__deepcopy__` and raise a NotImplemented exception there

Reply:
good suggestion to "overload __deepcopy__ and raise a NotImplemented exception there" for the time being

its also good suggestion to add a runtime check in `optimize`  (also for ask-tell interface), but that should be a small separate PR given that its more general

Reply:
Should be done now (see `gpflux/interface.py` and `test_interfaces.py` for gpflux)